tests/integration/test_information_reporting_interface.py

```diff
@@ -0,0 +1,60 @@
+"""Wakaama integration tests (pytest)"""
+import re
+import json
+import pytest
+
+
+# pylint: disable=unused-argument
+
+@pytest.mark.slow
+def test_observation_notification(lwm2mserver, lwm2mclient):
+    """LightweightM2M-1.1-int-301
+    Sending the observation policy to the device."""
+    # Test Procedure 1
+    assert lwm2mserver.commandresponse("read 0 /3/0/6", "OK")
+    text = lwm2mserver.waitforpacket()
+    assert text.find("COAP_205_CONTENT") > 0
+    packet = re.findall(r"(\[.*\])", text)
+    parsed = json.loads(packet[1])
+    assert next(item for item in parsed if item["n"] == "0")["v"] == 1 # Internal battery
+    assert next(item for item in parsed if item["n"] == "1")["v"] == 5 # USB
+    # Test Procedure 2
+    assert lwm2mserver.commandresponse("time 0 /3/0/7 5 15", "OK")
+    assert lwm2mserver.commandresponse("time 0 /3/0/8 10 20", "OK")
+    # Test Procedure 3
+    assert lwm2mserver.commandresponse("observe 0 /3/0/7", "OK")
+    assert lwm2mserver.commandresponse("observe 0 /3/0/8", "OK") #consumes first observe from /3/0/7
+    text = lwm2mserver.waitforpacket() # consumes ACK and piggy-backed first observe from /3/0/8
+    text = lwm2mserver.waitfortime(20) # wait for max "pmax"
+    # Pass-Criteria A
+    assert text.count("Notify from client #0 /3/0/7") > 0
+    assert text.count("Notify from client #0 /3/0/8") > 0
```

Comment:
Would it be feasible to check here also the values of Power Source Voltage (7) and Power Source Current (8)? Would be nice to check if the notification messages actually contain the observed values.

Reply:
Will add payload check